### PR TITLE
Fix h3dex corruption on re-asserting location

### PR DESCRIPTION
**Pull request by vihu**
- Refactor cases on assert location
- Allow siblings (N) to go to 0
- Delete existing h3dex and redo bootstrap_h3dex

### CODE REVIEW

File: src/blockchain.erl [hunk 1]
```diff
@@ -95,6 +95,7 @@
 -define(BC_UPGRADE_FUNS, [fun upgrade_gateways_v2/1,
                           fun bootstrap_hexes/1,
                           fun upgrade_gateways_oui/1,
+                          fun bootstrap_h3dex/1,
```
Comment (evanmcc) [inline src/blockchain.erl:98]: can we add a comment here noting that the duplication is not an error?

File: src/blockchain_hex.erl [hunk 2]
```diff
@@ -312,7 +312,7 @@ limit(Res, VarMap, OccupiedCount) ->
     DensityMax = maps:get(max, VarAtRes),
     DensityTgt = maps:get(tgt, VarAtRes),
     N = maps:get(n, VarAtRes),
-    Max = max((OccupiedCount - N), 1),
+    Max = max((OccupiedCount - N + 1), 1),
```
Comment (evanmcc) [inline src/blockchain_hex.erl:315]: moar parens

**Review by evanmcc — APPROVED** [review]
other than the nits I mentioned, looks good so far.

File: src/blockchain.erl (hunk 1 above)
Comment (vihu) [reply src/blockchain.erl:101]: added 10d662b

File: src/blockchain_hex.erl (hunk 2 above)
Comment (vihu) [reply src/blockchain_hex.erl]: added 10d662b

**Pull request merged**